modules/ROOT/pages/field-level-capabilities.adoc

```diff
@@ -0,0 +1,256 @@
+
+----
+public class ChatMultiLevelValueProvider implements ValueProvider {
+
+ @Connection
+ private ChatConnection chatConnection;
+
+ @Override
+ public Set<Value> resolve() {
+   Set<Value> values = new HashSet<>();
+   List<String> workspaces = chatConnection.getWorkspaces();
+   workspaces.stream().forEach(workspace -> {
+     ValueBuilder valueBuilder = newValue(workspace);
+     chatConnection.getChannels(workspace).stream().forEach(channel -> valueBuilder.withChild(newValue(channel)));
+     values.add(valueBuilder.build());
+   });
+   return values;
+ }
+
+ @Override
+ public String getId() {
+   return "Chat multilevel value provider";
+ }
+}
+----
+
+== Defining a Value Provider for XML Attributes
+
+Target selectors use the DataWeave syntax. If a parameter of an operation is an
+XML, you can use this syntax to define a value provider for an attribute that
+is a part of that parameter.
+
+The following example shows how the DataWeave syntax defines a value provider
+for an attribute:
+
+----
+public class ChatOperations {
+ public void sendChatMessage(@Connection ChatConnection chatConnection,
+      @FieldValues(value = ChannelIdValueProvider.class,  targetSelectors = "message.target.@channelId")
+      @TypeResolver(XmlChatMessageResolver.class) @Content InputStream body) {}
+}
+----
+
+In this example, the *body* parameter is an XML that has a root tag named
+*message*, which has a *target* that has the *channelId* attribute and an
+associated value provider.
+
+== Define Acting Parameters as Parts of Parameters
+
+To define acting parameters as parts of parameters, components must define
+bindings for their references for value providers and sample data providers.
+These bindings enable an acting parameter of the provider to take the value of
+the evaluated DataWeave expression.
+
+When no binding is defined for an acting parameter, the same implicit binding
+will be applied. The name of the acting parameter must be the same as the name
+of the parameter in the component to take that parameter’s value.
+
+Use the *@Binding* annotation to define an acting parameter. The *@FieldValues*
+annotation and the *@SampleData* annotation can be given an array of bindings
+to define the values the acting parameter must take.
+
+When a binding is defined, it declares the value an acting parameter will
+take by giving an extraction expression. This extraction expression is a
+DataWeave expression that uses only selectors to retrieve the value for the
+acting parameter.
+
+The available context for the expression has the value of all the component's
+parameters. The expression must start with the name of the parameter of the
+component. If the value comes from a part of the parameter, the expression
+follows the selector to the desired field.
+
+The following is an example where a parameter named *body* has a field with
+values that depend on the acting parameter *workspace*:
```

Comment:
```suggestion
values that depend on the acting parameter `workspace`:
```

Reply:
Since we are referring to code, not UI, use monospace. I may have missed some instances of this.